src/solvers/dgsem_tree/indicators_2d.jl

```diff
@@ -24,6 +24,105 @@ function create_cache(typ::Type{IndicatorHennemannGassner}, mesh, equations::Abs
+  # FV method is used in partially wet cells and guarantees the well-balanced property.
+  #
+  # Hard-coded cut-off value of `threshold_wet = 1e-4` was determined through many numerical experiments.
+  # Overall idea is to increase robustness when computing the velocity on (nearly) dry cells which
+  # could be "dangerous" due to division of conservative variables, e.g., v1 = hv1 / h.
+  # Here, the impact of the threshold on the number of cells being updated with FV is not that
+  # significant. However, its impact on the robustness is very significant.
+  # The value can be seen as a trade-off between accuracy and stability.
+  # Well-balancedness of the scheme on partially wet cells with hydrostatic reconstruction
+  # can only be proven for the FV method (see Chen and Noelle).
+  # Therefore we set alpha to be one regardless of its given value from the modal indicator.
+  threshold_wet = 1e-4
+
+  @threaded for element in eachelement(dg, cache)
+    indicator  = indicator_threaded[Threads.threadid()]
+    modal      = modal_threaded[Threads.threadid()]
+    modal_tmp1 = modal_tmp1_threaded[Threads.threadid()]
+
+    # (Re-)set dummy variable for alpha_dry
+    indicator_wet = 1
+
+    # Calculate indicator variables at Gauss-Lobatto nodes
+    for j in eachnode(dg), i in eachnode(dg)
+      u_local = get_node_vars(u, equations, dg, i, j, element)
+      h, _, _, _ = u_local
+
+      if h <= threshold_wet
+        indicator_wet = 0
+      end
+
+      indicator[i, j] = indicator_hg.variable(u_local, equations)
+    end
+
+    # Convert to modal representation
+    multiply_scalar_dimensionwise!(modal, dg.basis.inverse_vandermonde_legendre, indicator, modal_tmp1)
+
+    # Calculate total energies for all modes, without highest, without two highest
+    total_energy = zero(eltype(modal))
+    for j in 1:nnodes(dg), i in 1:nnodes(dg)
+      total_energy += modal[i, j]^2
+    end
+    total_energy_clip1 = zero(eltype(modal))
+    for j in 1:(nnodes(dg)-1), i in 1:(nnodes(dg)-1)
+      total_energy_clip1 += modal[i, j]^2
+    end
+    total_energy_clip2 = zero(eltype(modal))
+    for j in 1:(nnodes(dg)-2), i in 1:(nnodes(dg)-2)
+      total_energy_clip2 += modal[i, j]^2
+    end
+
+    # Calculate energy in higher modes
+    energy = max((total_energy - total_energy_clip1) / total_energy,
+                 (total_energy_clip1 - total_energy_clip2) / total_energy_clip1)
+
+    alpha_element = 1 / (1 + exp(-parameter_s / threshold * (energy - threshold)))
+
+    # Take care of the case close to pure DG
+    if alpha_element < alpha_min
+      alpha_element = zero(alpha_element)
+    end
+
+    # Take care of the case close to pure FV
+    if alpha_element > 1 - alpha_min
+      alpha_element = one(alpha_element)
+    end
+
+    # Clip the maximum amount of FV allowed or set to 1 depending on indicator_wet
+    alpha[element] = indicator_wet * min(alpha_max, alpha_element) - (indicator_wet - 1)
```

Comment:
What about a logic at least my human brain can parse easier, e.g.,
```suggestion
    # Clip the maximum amount of FV allowed or set to 1 depending on indicator_wet
    if indicator_wet
      alpha[element] = 1
    else
      alpha[element] = min(alpha_max, alpha_element)
    end
```
and choosing `Bool`ean values of the `indicator_wet` above?

Reply:
> I liked this version because it is compact

If you strongly prefer the current version, please go ahead. I prefer code that is easy to understand when looking at it - code is read much more often than it is written

Reply:
Of course, I do not want to stick to that idea if you do not like it. I will change it and use an if-clause.

@sloede thanks for your hint and sharing your experiences!